app/src/org/odk/collect/android/tasks/FormLoaderTask.java

```diff
@@ -329,28 +343,34 @@ public FormDef deserializeFormDef(File formDef) {
 
 
     /**
-     * Write the FormDef to the file system as a binary blog.
+     * Write the FormDef to the file system as a binary blob.
      * 
      * @param filepath path to the form file
+     * @throws IOException 
      */
-    public void serializeFormDef(FormDef fd, String filepath) {
-        // calculate unique md5 identifier
+    public void serializeFormDef(FormDef fd, String filepath) throws IOException {
+        // calculate unique md5 identifier for this form
         String hash = FileUtils.getMd5Hash(new File(filepath));
         File formDef = new File(Collect.CACHE_PATH + "/" + hash + ".formdef");
 
-        // formdef does not exist, create one.
+        // create a serialized form file if there isn't already one at this hash
         if (!formDef.exists()) {
-            FileOutputStream fos;
+            FileOutputStream fos = null;
             try {
                 fos = new FileOutputStream(formDef);
                 DataOutputStream dos = new DataOutputStream(fos);
                 fd.writeExternal(dos);
                 dos.flush();
                 dos.close();
-            } catch (FileNotFoundException e) {
-                e.printStackTrace();
-            } catch (IOException e) {
-                e.printStackTrace();
+            } finally {
+                //make sure we clean up the stream
+                if(fos != null) {
+                    try {
+                        fos.close();
+                    } catch (IOException e) {
+                        // there was already an error, this one is just cruft
```

Comment:
In this case we don't want to trump the exception getting thrown from fd.writeExternal. If, for instance, we threw a different type of exception (in the ticket it's a stack overflow), and then while cleaning up fos.close() errored out, we'll end up logging/reporting fos.close()'s exception path. 

It did just occur to me, however, that I shouldn't be closing dos above if we're closing fos in the finally cleanup, either way, I'll fix that